### PR TITLE
Module ADC rework

### PR DESCRIPTION
**Pull request by bzsolt**
Related issue: #1367

IoT.js-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

File: src/js/adc.js [hunk 1]
```diff
@@ -13,14 +13,16 @@
...
-    return new Adc();
-  }
-}
-
-Adc.prototype.open = function(configuration, callback) {
-  return new native.Adc(configuration, callback);
+var adc = {
+  open: function(config, callback) {
+    var adcPin = new native(config, function(err) {
+      callback(err, adcPin);
```
Comment (glistening) [inline src/js/adc.js]: @bzsolt What happens when user calls `open` without `callback` parameter?

Comment (bzsolt) [reply src/js/adc.js]: It will throw a TypeError.

Comment (glistening) [reply src/js/adc.js]: @bzsolt Although the behavior is changed from previous one, it seems reasonable. Also, I confirmed node also throws `TypeError`.

```
$ node -e "require('fs').open('/etc/hosts','r')"
fs.js:131
    throw new TypeError('"callback" argument must be a function');
    ^
```

**Comment by yichoi** [comment]
@bzsolt rebase please

**Comment by bzsolt** [comment]
@yichoi The PR is rebased.

**Pull request merged**